src/main/kotlin/pl/put/smartgarden/infra/api/DeviceController.kt

```diff
@@ -0,0 +1,59 @@
+package pl.put.smartgarden.infra.api
+
+import io.swagger.annotations.Api
+import io.swagger.annotations.ApiOperation
+import io.swagger.annotations.ApiResponse
+import io.swagger.annotations.ApiResponses
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import pl.put.smartgarden.domain.device.Device
+import pl.put.smartgarden.domain.device.DeviceFacade
+import pl.put.smartgarden.domain.device.Measure
+import pl.put.smartgarden.domain.device.dto.DeviceDto
+import pl.put.smartgarden.domain.device.dto.DeviceMeasuresDto
+
+@Api(description = "Devices api")
+@RestController
+@RequestMapping("/devices")
+class DeviceController(
+    val deviceFacade: DeviceFacade
+) {
+    @PutMapping
+    @ApiOperation("Create or update device.")
+    @ApiResponses(value = [
+        ApiResponse(code = 200, message = "OK"),
+        ApiResponse(code = 400, message = "Bad request"),
+        ApiResponse(code = 409, message = "Conflict")
+    ])
+    @ResponseStatus(HttpStatus.OK)
+    fun createOrUpdateDevice(@RequestBody deviceDto: DeviceDto): Device = deviceFacade.createOrUpdateDevice(deviceDto)
+
+    @PostMapping("/measures")
```

Comment:
Myślę że lepiej byłoby np. "/{id}/measures", po pierwsze będzie to czytelniejszy endpoint, a po drugie musimy skądś brać informację o które urządzenie chodzi, wydaje mi się że informacja o tym powinna być właśnie w Path, a nie w body.

Reply:
to nadal zależy od tego jak będziemy uwierzytelniać to raspberry, ale jeśli bez tokena, to faktycznie tu lepiej niż w body